logical/src/main/java/org/apache/drill/common/expression/PathSegment.java

```diff
@@ -151,6 +151,15 @@ public boolean isNamed() {
       return true;
     }
 
+    /**
+     * Checks that the path of this name segment is complex.
+     *
+     * @return true if the path of this name segment contains dots
+     */
+    public boolean isComplex() {
```

Comment:
Not sure I agree with this. We just did changes to enforce the rule that a path segment is represented by a distinct object. If a path segment can contain dots, then we don't need to represent parts as multiple parts. But, we went the multiple part route to allow dots in names.

How will we know if a dot in a name represents a complex (multi-part) name vs. a simple name that happens to contain a dot?

In short, this seems a bit of a hack and introduces undesirable ambiguity.